launch/nav.launch

```diff
@@ -0,0 +1,72 @@
+  <node unless="$(arg gui)" name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
+  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
+  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
+
+  <include file="$(find gazebo_ros)/launch/empty_world.launch">
+    <env name="GAZEBO_MODEL_PATH" value="$(find mrover)/env_description" />
+    <arg name="world_name" value="$(find mrover)/env_description/world.world" />
+    <arg name="paused" value="false" />
+    <arg name="use_sim_time" value="true" />
+    <arg name="gui" value="true" />
+    <arg name="headless" value="false" />
+    <arg name="debug" value="false" />
+    <arg name="verbose" default="false" />
+    <arg name="gui_required" value="true" />
+  </include>
+
+  <node name="spawn_model" pkg="gazebo_ros" type="spawn_model" output="screen" args="
+    -urdf -param robot_description
+    -model rover" />
+
+  <!-- Perception-->
+  <node name="aruco_detect" pkg="mrover" type="aruco_detect">
+    <param name="publish_images" value="true" />
+    <param name="fiducial_len" value="0.20" />
+    <param name="dictionary" value="0" /> <!-- DICT_4X4_50 -->
+    <remap from="camera" to="camera/color/image_raw" />
+    <remap from="camera_info" to="camera/color/camera_info" />
+  </node>
+
+  <!-- Navigation -->
+  <!-- This publishes the transforms of the course waypoints -->
+  <node name="debug_course_publisher" pkg="mrover" type="debug_course_publisher.py" />
+
+  <!--
+    ============
+    Localization
+    ============
+    Frames in the free tree:
+    odom: The topmost global frame
+    base_link: The rover, 
+  -->
+  <node name="mrover_localization" pkg="mrover" type="localization.py" />
+
+  <rosparam command="load" file="$(find mrover)/params/ekf_localization.yaml" />
+
+  <!-- Fuses together GPS position and IMU orientation with an Extended Kalman Filter (EKF) -->
+  <!-- Publishes a resulting tf entry between odom and base_link -->
+  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_odom" clear_params="true">
+    <!-- Publications -->
+    <remap from="odometry/filtered" to="odometry/filtered" />
+  </node>
+
+  <!-- Reads the GPS topic and convert it to cartesian based on a reference point-->
+  <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform" clear_params="true">
+    <!-- Reference latitude/longitude to linearize from -->
+    <rosparam param="datum">[42.2, -83.7, 0.0, map, base_link]</rosparam>
+    <!-- Subscriptions -->
+    <remap from="imu/data" to="imu" />
+    <remap from="gps/fix" to="gps/fix" />
+    <remap from="odometry/filtered" to="odometry/filtered" />
+    <!-- Publications -->
+    <remap from="gps/filtered" to="gps/filtered" />
+    <remap from="odometry/gps" to="odometry/gps" />
```

Comment:
???

Reply:
I just left it in to show what topics were published/subscribed (instead of having to go look at docs), and in case they need to be remapped in the future.

Reply:
I see, maybe add a comment just mentioning that some of them are mapped to themselves just for visibility

Reply:
Kk